File: 0590-n-ary-tree-postorder-traversal/0590-n-ary-tree-postorder-traversal.java
/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, List<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
    public List<Integer> postorder(Node root) {
        //Create stack and return answer variables.
        LinkedList<Integer> ans = new LinkedList<>();
        //Handling root == null case
        if(root==null) return ans;
        // add in stack root 
        LinkedList<Node> st = new LinkedList<>();
        st.addLast(root);
        // keep adding removenode children in stack
        while(st.size()!=0) {
            Node removeN = st.removeLast();
            
            for(Node child : removeN.children) {
                st.addLast(child);
            }
            // keep adding remove Node at 0th idx in our ans;
            ans.addFirst(removeN.val);
        }
        
        return ans;
    }
}

File: 0590-n-ary-tree-postorder-traversal/0590-n-ary-tree-postorder-traversal.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    vector<int> postorder(Node* root) {
        vector<int> ans;
        if(root == nullptr) return ans;
        
        stack<Node*> st;
        st.push(root);
        
        while(st.size()!=0) {
            Node* removeN = st.top(); st.pop();
            
            for(Node* child : removeN->children) {
                st.push(child);
            }
            
            ans.insert(ans.begin(),removeN->val);
        }
        
        return ans;
    }
};